src/main/kotlin/com/moshbit/katerbase/MongoDatabase.kt

```diff
@@ -227,9 +293,15 @@ open class MongoDatabase(
       }
 
       thread(isDaemon = true, name = "katerbase-watch-${name}-${Random().nextInt().absoluteValue}") {
+        fun watch() = if (session != null) {
+          internalCollection.watch(session, listOf(pipeline))
+        } else {
+          internalCollection.watch(listOf(pipeline))
+        }
```

Comment:
why always use 2 separate function calls in this PR? you can always pass the session as first argument. (here and everywhere)

Reply:
yes, that would be much nicer, sadly there is a `notNull(..)` check on it, i can't pass a `null` here. 
In the future, if we always use a session for every mongo client that we create (which is recommended), we could clean this up